x/oracle/keeper/task.go

```diff
@@ -202,6 +205,9 @@ func (k Keeper) CreateTask(ctx sdk.Context, creator sdk.AccAddress, task types.T
 		if err := k.CollectBounty(ctx, task.GetBounty(), creator); err != nil {
 			return err
```

Comment:
What does this task happen if collectBounty executes failed?  This task is already saved.

Reply:
If `collectBounty` fails, the msg execution will fail and the db operations that occurred before the failure will not be 'submitted' to DB. In a nutshell, data consistency will not be affected.